From afa1b1ec11ed3e83a05168cf9489c5e8dcdaebf7 Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Fri, 29 Sep 2023 14:54:43 +0530
Subject: [PATCH 069/120] usb: dwc3: Modify generic_phy_set_mode call to pass
 usb_role info

Pass usb_role info to generic_phy_set_mode call for phy
to correctly handle usb role

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/u-boot.git 5a33716452c3b1e0dc982607de776c81f1303f3e usb: dwc3: Modify generic_phy_set_mode call to pass usb_role info]
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Change-Id: I687f19e499481c2be529fdc6a712770d663390f0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/u-boot/+/330523
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Patrice CHOTARD <patrice.chotard@foss.st.com>
---
 drivers/usb/dwc3/core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index 7f90eeb7648..3b951752fa9 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -1043,7 +1043,8 @@ int dwc3_setup_phy(struct udevice *dev, struct phy_bulk *phys)
 			goto err_mode;
 	}
 
-	ret = generic_phy_set_mode_bulk(phys, phymode, 0);
+	ret = generic_phy_set_mode_bulk(phys, phymode, (dr_mode == USB_DR_MODE_HOST) ?
+					USB_ROLE_HOST : USB_ROLE_DEVICE);
 	if (ret)
 		goto err_mode;
 
-- 
2.47.2

